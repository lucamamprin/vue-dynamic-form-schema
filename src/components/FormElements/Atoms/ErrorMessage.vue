<template>
    <span
      v-html="errorMessage"
      :id="id"
      class="help m-0"
      :class="`${inlineClasses} ${positionClass} d-${display}`"
    >
    </span>
</template>

<script>
export default {
  computed: {
    positionClass () {
      return this.position.length ? `position-${this.position}` : ""
    },
    inlineClasses () {
      return this.inlineError ? "ml-1" : ""
    },
  },
  props: {
    errorMessage: String,
    id: {
      type: String,
      required: true,
    },
    // should the error be displayed with a left margin rather than a top margin?
    inlineError: {
      type: Boolean,
      default: false,
    },
    position: {
      type: String,
      default: "",
    },
    display: {
      type: String,
      default: "inline-block",
    },
  },
}
</script>
