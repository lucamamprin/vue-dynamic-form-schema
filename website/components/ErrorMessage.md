# ErrorMessage

## Props

<!-- @vuese:ErrorMessage:props:start -->
|Name|Description|Type|Required|Default|
|---|---|---|---|---|
|errorMessage|-|`String`|`true`|-|
|id|-|`String`|`true`|-|
|inlineError|should the error be displayed with a left margin rather than a top margin?|`Boolean`|`false`|false|
|position|-|`String`|`false`|-|
|display|-|`String`|`false`|inline-block|

<!-- @vuese:ErrorMessage:props:end -->


