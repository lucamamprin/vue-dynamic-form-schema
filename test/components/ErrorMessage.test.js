import Vue from "vue"
import { shallowMount } from "@vue/test-utils"
import ErrorMessage from "../../src/components/FormElements/Atoms/ErrorMessage.vue"

const propsDefault = {
  errorMessage: "Generic error message",
  id: "error",
}

const positions = [
  "static",
  "absolute",
  "relative",
]

// BootStrap display classes, to be deprecated on next major refactor
const bsDisplay = [
  "inline",
  "block",
]

let wrapper

describe("error message", () => {

  beforeEach(() => {
    wrapper = shallowMount(ErrorMessage, {
      propsData: propsDefault,
    })
  })

  test("mounts correctly", () => {
    expect(wrapper).toMatchSnapshot()
  })

  test("prints inline error class", async () => {
    await wrapper.setProps({
      inlineError: true,
    })

    await Vue.nextTick()

    expect(wrapper).toMatchSnapshot()
  })

  positions.forEach(position => {
    test(`prints ${position} position class`, async () => {
      await wrapper.setProps({
        position,
      })

      await Vue.nextTick()

      expect(wrapper).toMatchSnapshot()
    })
  })

  bsDisplay.forEach(display => {
    test(`prints ${display} position class`, async () => {
      await wrapper.setProps({
        display,
      })

      await Vue.nextTick()

      expect(wrapper).toMatchSnapshot()
    })
  })
})
