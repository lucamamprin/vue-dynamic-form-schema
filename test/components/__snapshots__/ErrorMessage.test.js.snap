// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`error message mounts correctly 1`] = `
<span id="error" class="help m-0   d-inline-block">
  Generic error message
</span>
`;

exports[`error message prints absolute position class 1`] = `
<span id="error" class="help m-0  position-absolute d-inline-block">
  Generic error message
</span>
`;

exports[`error message prints block position class 1`] = `
<span id="error" class="help m-0   d-block">
  Generic error message
</span>
`;

exports[`error message prints inline error class 1`] = `
<span id="error" class="help m-0 ml-1  d-inline-block">
  Generic error message
</span>
`;

exports[`error message prints inline position class 1`] = `
<span id="error" class="help m-0   d-inline">
  Generic error message
</span>
`;

exports[`error message prints relative position class 1`] = `
<span id="error" class="help m-0  position-relative d-inline-block">
  Generic error message
</span>
`;

exports[`error message prints static position class 1`] = `
<span id="error" class="help m-0  position-static d-inline-block">
  Generic error message
</span>
`;
